### DISCUSSION TITLE
Repeat by Variable - Repeat QUERY within the same panel

### DISCUSSION THREAD

**Discussion by ebachman24** [grafana/grafana#60527] [Dashboards, Panels & Visualizations]
In panel options, when enabling repeat by variable, other panels are automatically generated based on how many variables are selected.  4 variables selected, 4 panels are automatically generated (you probably know how it works).
I'm wondering if it would be possible to include an option to repeat the query WITHIN the same panel. So instead of automatically generating other panels, new queries would automatically by generated.  i.e. 4 variables selected, 4 queries automatically generated in the same panel (A, B, C, D).  

This would be very useful if there are use cases where math or other transformations need to be done on different queries - controllable through variables (multi-value usually).

*Note: this would also require the ability to put the variable value in the title of the queries. Or at least something similar to keep track of which is which.  

![image](https://user-images.githubusercontent.com/104096952/208451594-f96e7c57-945a-4307-a1b8-55201133d27b.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
